core/src/main/java/org/jdbi/v3/core/ConfigOnlyExtension.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core;
+
+import org.jdbi.v3.core.extension.ExtensionConfig;
+import org.jdbi.v3.core.extension.ExtensionFactory;
+
+class ConfigOnlyExtension<C extends ExtensionConfig<C>> implements ExtensionFactory<C>
```

Comment:
This is the thing I'm most curious for review of.  The intention here is to provide an extensible way for *core* jdbi to customize, e.g. cache sizes or other "indirect" tuneables, in this case "sql array binding strategy".

Other option would be to have a `ObjectArrayArgumentFactory` that binds `Object[]` instead of `sql.Array` for H2's benefit.  Might be simpler overall, but I think the concept here is interesting.